merkledag/merkledag.go

```diff
@@ -269,3 +313,24 @@ func (np *nodePromise) Get(ctx context.Context) (*Node, error) {
 	}
 	return np.cache, nil
 }
+
+// EnumerateChildren will walk the dag below the given root node and add all
+// unseen children to the passed in set.
+// TODO: parallelize to avoid disk latency perf hits?
```

Comment:
this is going to be used by more than just GC. 

and remember during GC, no adds can be made.

i'm ok merging this and improving it later, but i'm pretty sure we're going to be annoyed by it.  also, we already have concurrent fetch elsewhere-- i imagine it can be reused here.